### PR TITLE
Euler eToken oracle

### PR DESCRIPTION
**Pull request by davidbrai**
Adding an oracle for Euler ETokens to be able to use ETokens as collateral.

Uses `EToken.convertBalanceToUnderlying` and `EToken.convertUnderlyingToBalance` for price conversion.

### CODE REVIEW

File: packages/foundry/contracts/oracles/euler/IEToken.sol [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0;
+import "@yield-protocol/utils-v2/contracts/token/IERC20Metadata.sol";
+
+interface IEToken {
+    function convertBalanceToUnderlying(uint256 balance) external view returns (uint256);
```
Comment (iamsahu) [inline packages/foundry/contracts/oracles/euler/IEToken.sol:6]: Copy natspec from EToken?

**Review by iamsahu — COMMENTED** [review]
Apart from a minor Natspec request, LGTM

File: packages/foundry/contracts/oracles/euler/IEToken.sol (hunk 1 above)
Comment (davidbrai) [reply packages/foundry/contracts/oracles/euler/IEToken.sol:10]: 👍 done

**Review by iamsahu — APPROVED** [review]
Great job David! 🚀

**Pull request merged**